### PR TITLE
Contrast 27048

### PR DESCRIPTION
**Pull request by serge17**
Now, plugin uses application id instead of name to query vulnerabilities.
Plugin is not compatible with its previous version, so the global configuration, pipelines, and post-build actions need to be updated.

post-build action

pipeline syntax

untitled

### CODE REVIEW

**Comment by gmoore** [comment]
1) Change the display in the drop down to be : Application name (id)

2) Let's discuss ways to make this backwards compatible. I think we can pull it off.

**Comment by serge17** [comment]
@gmoore Post-build action should be compatible now. Should the users be able to use pipeline from the previous version as well?

**Comment by gmoore** [comment]
Not sure what you mean by "Should the users be able to use pipeline from the previous version as well?"

**Comment by serge17** [comment]
@gmoore The compatibility issues with the previous plugin version should be resolved now. Can you check again please? I will work on fixing the tests.

**Comment by serge17** [comment]
@gmoore Tests should be fixed now

File: src/main/java/com/aspectsecurity/contrast/contrastjenkins/VulnerabilityTrendRecorder.java [hunk 1]
```diff
@@ -51,8 +49,39 @@ public VulnerabilityTrendRecorder(List<ThresholdCondition> conditions, String te
         this.queryBy = queryBy;
     }
 
-    public TeamServerProfile getProfile() {
-        return VulnerabilityTrendHelper.getProfile(teamServerProfileName);
+    protected Object readResolve() {
```
Comment (gmoore) [inline src/main/java/com/aspectsecurity/contrast/contrastjenkins/VulnerabilityTrendRecorder.java:52]: @serge17 Can you explain a little bit what's going on here?

File: src/main/java/com/aspectsecurity/contrast/contrastjenkins/VulnerabilityTrendStep.java [hunk 2]
```diff
@@ -146,11 +154,28 @@ public Step newInstance(Map<String, Object> arguments) {
...
+
+                if (applicationId != null) {
+                    step.setApplicationId((String) applicationId);
+                } else {
+                    throw new IllegalArgumentException("Application ID must be set.");
+                }
+
+            }
+
+            ////
```
Comment (gmoore) [inline src/main/java/com/aspectsecurity/contrast/contrastjenkins/VulnerabilityTrendStep.java]: You can take out these little guys `////`

File: src/main/java/com/aspectsecurity/contrast/contrastjenkins/VulnerabilityTrendRecorder.java (hunk 1 above)
Comment (serge17) [reply src/main/java/com/aspectsecurity/contrast/contrastjenkins/VulnerabilityTrendRecorder.java:52]: This method is often used in Jenkins plugins to migrate data from one field to another when the field is deprecated. We added new variables "apps" and "applicationId" to profile, and we populate them in this method. It is triggered automatically by Jenkins. 
https://wiki.jenkins.io/display/JENKINS/Hint+on+retaining+backward+compatibility

**Comment by gmoore** [comment]
@serge17 Give this another test and get a release up ASAP. Also take care of the other stories in the epic.

**Pull request merged**